pandas/core/frame.py

```diff
@@ -6237,13 +6237,31 @@ def explode(self, column: Union[str, Tuple]) -> "DataFrame":
         if not self.columns.is_unique:
             raise ValueError("columns must be unique")
 
-        result = self[column].explode()
-        return (
+        if isinstance(self.index, MultiIndex):
+            index_names = [
+                "level_{num}".format(num=num) if val is None else val
+                for num, val in enumerate(self.index.names)
```

Comment:
I think you don't care about the names if you set the index after the join with something like:
```python
result.index = pandas.MultiIndex.from_frame(result.iloc[:, :self.index.nlevels])
```

I think this will make this code a bit simpler.

Reply:
I may have misunderstood, but are you sure we can disregard names? If the original index is unnamed, then after `reset_index()`, it will be named `index`. 

E.g.:
```
import pandas as pd
df = pd.DataFrame([1,2,3]).reset_index()
df.iloc[:, :df.index.nlevels]
```
>   index
>0      0
>1      1
>2      2

Reply:
You're setting the index at the very end to the original one, you'll have to leave that.

Reply:
I've pushed a simpler solution which uses the change you suggested - thanks for taking the time to review my work!